tests/v2/test_waku_noise.nim

```diff
@@ -413,3 +412,438 @@ procSuite "Waku Noise":
+    var aliceHS = initialize(hsPattern, staticKey = aliceStaticKey, initiator = true)
+
+    let bobStaticKey = genKeyPair(rng[])
+    var bobHS = initialize(hsPattern, staticKey = bobStaticKey)
+    
+    var 
+      sentTransportMessage: seq[byte]
+      aliceStep, bobStep: HandshakeStepResult 
+    
+    # Here the handshake starts
+    # Write and read calls alternate between Alice and Bob: the handhshake progresses by alternatively calling stepHandshake for each user
+
+    ###############
+    # 1st step
+    ###############
+
+    # We generate a random transport message
+    sentTransportMessage = randomSeqByte(rng[], 32)
+
+    # By being the handshake initiator, Alice writes a Waku2 payload v2 containing her handshake message 
+    # and the (encrypted) transport message
+    aliceStep = stepHandshake(rng[], aliceHS, transportMessage = sentTransportMessage).get()
+
+    # Bob reads Alice's payloads, and returns the (decrypted) transport message Alice sent to him
+    bobStep = stepHandshake(rng[], bobHS, readPayloadV2 = aliceStep.payload2).get()
+    
+    check:
+      bobStep.transportMessage == sentTransportMessage
+
+    ###############
+    # 2nd step
+    ###############
+    
+    # We generate a random transport message
+    sentTransportMessage = randomSeqByte(rng[], 32)
+
+    # At this step, Bob writes and returns a payload
+    bobStep = stepHandshake(rng[], bobHS, transportMessage = sentTransportMessage).get()
+
+    # While Alice reads and returns the (decrypted) transport message
+    aliceStep = stepHandshake(rng[], aliceHS, readPayloadV2 = bobStep.payload2).get()
+    
+    check: 
+      aliceStep.transportMessage == sentTransportMessage
+
+    ###############
+    # 3rd step
+    ###############
+
+    # We generate a random transport message
+    sentTransportMessage = randomSeqByte(rng[], 32) 
+
+    # Similarly as in first step, Alice writes a Waku2 payload containing the handshake message and the (encrypted) transport message
+    aliceStep = stepHandshake(rng[], aliceHS, transportMessage = sentTransportMessage).get()
+
+    # Bob reads Alice's payloads, and returns the (decrypted) transport message Alice sent to him
+    bobStep = stepHandshake(rng[], bobHS, readPayloadV2 = aliceStep.payload2).get()
+    
+    check: 
+      bobStep.transportMessage == sentTransportMessage
+
+    # Note that for this handshake pattern, no more message patterns are left for processing 
+    # Another call to stepHandshake would return an empty HandshakeStepResult
+    # We test that extra calls to stepHandshake do not affect handshake finalization
+    bobStep = stepHandshake(rng[], bobHS, transportMessage = sentTransportMessage).get()
+    aliceStep = stepHandshake(rng[], aliceHS, readPayloadV2 = bobStep.payload2).get()
+    aliceStep = stepHandshake(rng[], aliceHS, transportMessage = sentTransportMessage).get()
+    bobStep = stepHandshake(rng[], bobHS, readPayloadV2 = aliceStep.payload2).get()
+   
```

Comment:
As per our call, please add an explicit check to  whether the handshake states are remained unchanged or not. The same for the hand shake results.

Reply:
Addressed in https://github.com/status-im/nwaku/pull/934/commits/1d86c99113c3a79e07892e5cf8821baf104845a3